scio-core/src/main/java/com/spotify/scio/transforms/BaseAsyncLookupDoFn.java

```diff
@@ -165,16 +151,21 @@ public void processElement(ProcessContext c, BoundedWindow window) {
             },
             throwable -> {
               semaphore.release();
-              results.add(new Result(input, failure(throwable), c.timestamp(), window));
-              futures.remove(uuid);
+              results.add(new Result(input, failure(throwable), key, c.timestamp(), window));
```

Comment:
Should `futures.remove(key)` here too?

Reply:
In the LookupDoFn family errors are actually passed through to the consumer by wrapping them in a `Try`, so this doesn't indicate failure of the DoFn.

In this case the normal `flush()` calls will remove the futures as if everything went correctly